github/client_test.go

```diff
@@ -0,0 +1,382 @@
+	t.Parallel()
+	logger := testhelpers.Logger(t)
+
+	// Create a mock transport
+	mockTransport := &mockRoundTripper{
+		responses: make(map[string]*http.Response),
+	}
+
+	client, err := NewClient(logger, "test-token", mockTransport)
+
+	require.NoError(t, err)
+	assert.NotNil(t, client)
+	assert.NotNil(t, client.Rest)
+	assert.NotNil(t, client.GraphQL)
+}
+
+func TestClientRoundTripper(t *testing.T) {
+	t.Parallel()
+	logger := testhelpers.Logger(t)
+
+	// Test with nil next transport (should use default)
+	transport := clientRoundTripper("TEST", logger, nil)
+	assert.NotNil(t, transport)
+	assert.IsType(t, &loggingTransport{}, transport)
+
+	// Test with custom next transport
+	mockTransport := &mockRoundTripper{
+		responses: make(map[string]*http.Response),
+	}
+	transport = clientRoundTripper("TEST", logger, mockTransport)
+	assert.NotNil(t, transport)
+	assert.IsType(t, &loggingTransport{}, transport)
+
+	lt := transport.(*loggingTransport)
+	assert.Equal(t, "TEST", lt.clientType)
+	assert.Equal(t, mockTransport, lt.transport)
+}
+
+func TestLoggingTransportRoundTrip(t *testing.T) {
+	t.Parallel()
+	logger := testhelpers.Logger(t)
+
+	tests := []struct {
+		name          string
+		statusCode    int
+		headers       map[string]string
+		expectError   bool
+		isRateLimited bool
+		isMockRequest bool
+	}{
+		{
+			name:       "successful request",
+			statusCode: http.StatusOK,
+			headers: map[string]string{
+				"X-RateLimit-Remaining": "4999",
+				"X-RateLimit-Limit":     "5000",
+				"X-RateLimit-Used":      "1",
+				"X-RateLimit-Reset":     fmt.Sprintf("%d", time.Now().Add(time.Hour).Unix()),
+			},
+		},
+		{
+			name:       "rate limit warning",
+			statusCode: http.StatusOK,
+			headers: map[string]string{
+				"X-RateLimit-Remaining": "50",
+				"X-RateLimit-Limit":     "5000",
+				"X-RateLimit-Used":      "4950",
+				"X-RateLimit-Reset":     fmt.Sprintf("%d", time.Now().Add(time.Hour).Unix()),
+			},
+		},
+		{
+			name:       "mock request (no warning)",
+			statusCode: http.StatusOK,
+			headers: map[string]string{
+				"X-RateLimit-Remaining": "50",
+				"X-RateLimit-Limit":     "5000",
+				"X-RateLimit-Used":      "4950",
+				"X-RateLimit-Reset":     fmt.Sprintf("%d", time.Now().Add(time.Hour).Unix()),
+			},
+			isMockRequest: true,
+		},
+		{
+			name:          "rate limited",
+			statusCode:    http.StatusTooManyRequests,
+			isRateLimited: true,
+		},
+		{
+			name:       "missing headers (defaults to 0)",
+			statusCode: http.StatusOK,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			mockTransport := &mockRoundTripper{
+				responses: make(map[string]*http.Response),
+			}
+
+			transport := &loggingTransport{
+				transport:  mockTransport,
+				logger:     logger,
+				clientType: "TEST",
+			}
+
+			// Create test server
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
If you're creating a server here, what's the point of baking in the mock request functionality?